packages/db/postgres/functions.sql

```diff
@@ -251,17 +251,37 @@ BEGIN
 END
 $$;
 
-CREATE OR REPLACE FUNCTION user_used_storage(query_user_id BIGINT) RETURNS TEXT
-  LANGUAGE plpgsql
+CREATE TYPE used_storage AS (uploaded TEXT, pinned TEXT);
+
+CREATE OR REPLACE FUNCTION user_used_storage(query_user_id BIGINT) 
+RETURNS used_storage
+LANGUAGE plpgsql
 AS
 $$
+DECLARE used_storage used_storage;
 BEGIN
-  return(
-    select sum(c.dag_size)
-    from upload u
-    join content c on c.cid = u.content_cid
-    where u.user_id = query_user_id and u.deleted_at is null
-  )::TEXT;
+  SELECT COALESCE(SUM(c.dag_size), 0)
+  INTO used_storage.uploaded::TEXT
+  FROM upload u
+  JOIN content c ON c.cid = u.content_cid
+  WHERE u.user_id = query_user_id::BIGINT
+  AND u.deleted_at is null;
+
+  SELECT COALESCE((
+    SELECT SUM(c.dag_size)
+    FROM psa_pin_request pr
+    JOIN pin p ON p.content_cid = pr.content_cid
```

Comment:
nit.
I know it doesn't change, and we could leave this as is... but the order of the joins here is bothering me.
To map the fk relations should be psa_pin_request JOIN content JOIN pin.

Reply:
Changed to map the fk relationships.

Reply:
Thanks and sorry for being annoying!